### PR TITLE
Add goto-harness man page

### PR DESCRIPTION
**Pull request by tautschnig**
This was built using help2man and then manually expanded. In the process
of preparing it, several errors in existing goto-harness documentation
were found and fixed as well.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-harness/memory_snapshot_harness_generator.cpp [hunk 1]
```diff
@@ -111,7 +111,7 @@ void memory_snapshot_harness_generatort::validate_options(
   {
     throw invalid_command_line_argument_exceptiont(
       "option --memory_snapshot is required",
-      "--harness-type initialise-from-memory-snapshot");
+      "--harness-type initialise-with-memory-snapshot");
```
Comment (peterschrammel) [inline src/goto-harness/memory_snapshot_harness_generator.cpp]: `initialise-with-memory-snapshot` -> `initialize-with-memory-snapshot` (to be consistent with AE spelling we use everywhere else) 🇺🇸

File: src/goto-harness/memory_snapshot_harness_generator_options.h [hunk 2]
```diff
@@ -33,7 +33,7 @@ Author: Diffblue Ltd.
 // clang-format off
 #define MEMORY_SNAPSHOT_HARNESS_GENERATOR_HELP                                 \
   "memory snapshot harness generator (--harness-type\n"                        \
-  "  initialise-from-memory-snapshot)\n\n"                                     \
+  "  initialise-with-memory-snapshot)\n\n"                                     \
```
Comment (peterschrammel) [inline src/goto-harness/memory_snapshot_harness_generator_options.h]: 🇺🇸

**Review by peterschrammel — APPROVED** [review]
For some reason Github thinks .1 files are not part of the diff and refuses inline comments.
`First,function-call` -> `First, function-call`
initialise 🇺🇸

File: src/goto-harness/memory_snapshot_harness_generator.cpp (hunk 1 above)
Comment (tautschnig) [reply src/goto-harness/memory_snapshot_harness_generator.cpp]: I've added a separate commit fixing this across code, documentation, tests.

**Pull request merged**